### DISCUSSION TITLE
Running Wavelog on Windows/XAMPP ?

### DISCUSSION THREAD

**Discussion by otalado** [wavelog/wavelog#628] [Q&A] [answered]
I encountered some errors while trying to install Wavelog in Win11 and XAMPP, both the latest versions. When I started the installation, I came across an error caused by line 101 of ...xampp\htdocs\Wavelog\install\index.php:
`setcookie($gt_conf['lang_cookie'], $browser_language['gettext']);`
It seemed that 
`$browser_language['gettext']` 
was empty. To fix this, I replaced that line with:
`setcookie($gt_conf['lang_cookie'], "english");`
After making this change, I was able to continue the installation. I also received a warning about `max_execution_time`, and although I tried changing it as suggested, the warning persisted. I left it as it was and continued with the installation. However, when I reached the Finish stage, I encountered an error in file ...\xampp\htdocs\Wavelog\install\includes\database_class.php line 55 
(`$mysqli->multi_query($newquery);`).

At this point, I decided to give up. Now, I have a few questions:
- I am using PHP version 8.2.12. Do you think the issues I encountered could be due to the version I have not being supported?
- If anyone has successfully installed Wavelog on XAMPP, could you please provide information regarding the versions of PHP and MariaDB they used?

**Comment by HB9HIL** [comment]
Hello @otalado 

- the issue with `$browser_language`: This one is known and fixed in #603
- when you changed `max_execution_time` in php.ini you have to restart the webserver
- regarding the last point maybe @AndreasK79 can help, as he uses XAMPP aswell

But please hang tight, fixes for the installer are in current progress in the PR mentioned above

**Comment by AndreasK79** [comment]
I've been using Xampp for several years without any trouble. I have not seen the issue with $mysqli->multi_query($newquery);. I installed Wavelog today when testing the new installer.

I use PHP 8.2.4 and MariaDB 10.4.13. Did it produce any more info regarding the multi_query error?

**Reply by otalado** [reply]
Thank you both HB9HIL and AndreasK97 for the replies. I found the version of XAMPP which uses PHP 8.2.4 but I could not find the version using the MariaDB 10.4.13 but the **version 10.4.28**  I installed that version, then tried to install the Wavelog . Installation works until the Finish phase, where it halted with:
`Fatal error: Uncaught mysqli_sql_exception: MySQL server has gone away in D:\xampp\htdocs\Wavelog\install\includes\database_class.php:55 Stack trace: #0 D:\xampp\htdocs\Wavelog\install\includes\database_class.php(55): mysqli->multi_query('SET FOREIGN_KEY...') #1 D:\xampp\htdocs\Wavelog\install\index.php(175): Database->create_tables(Array) #2 {main} thrown in D:\xampp\htdocs\Wavelog\install\includes\database_class.php on line 55`
So, the error is triggered by the msqli. That may be attributed to the MariaDB version (I think not very likely) or to my php.ini configuration. If it is the latter it would be very helpful if AndreasK79 could provide me with his version of the php.ini so that I can compare both.
I also have the LAMP stack on my other Win PC which I built myself (i.e. not XAMPP or WAMP or such). I'll back it up and then test the Wavelog on it and see what happens...

**Comment by AndreasK79** [comment]
@otalado my xampp is a few years old. I have just updated php manually. I am also running Win10, not that I know if it makes any difference. Here is my php.ini: https://pastebin.com/1mdjqjrV

**Comment by AndreasK79** [comment, marked as answer]
@HB9HIL managed to track down the problem, and I could reproduce it myself.

@otalado you need to open my.ini and set max_allowed_packet=10M instead of 1M under the mysqld section. Then restart the database.
The reason is that the install.sql file is quite big, but it'll be reworked to be smaller.

**Comment by otalado** [comment]
@AndreasK79: Thanks. I also found a suggestion for this issue: 
`https://stackoverflow.com/questions/7942154/mysql-error-2006-mysql-server-has-gone-away`
I tried the setting with the following:
`max_allowed_packet=16M`
as suggested there. That solved the issue but I did no further tests as I had other stuff to do. Now that I have returned, I found your reply which suggests almost the same thing. Since you (obviously) have better in-depth knowledge of Wavelog, I decided to follow your advice and set the max_allowed_packet to 10M.
By the way - I ditched the XAMPP and installed the stack from zipped packages, so if you keep the list of working configs:
**Apache 2.4.32** - 240718 (from Apache lounge)
**PHP 8.2.4**
**MariaDB 11.4.2**
and I don't use phpMyAdmin as I prefer things like MySQL Workbench or HeidiSQL quicker, easier to use, and more powerful tools.
So @HB9HIL and @AngreasK79 thank you both for your help!

**Reply by AndreasK79** [reply]
@otalado great to hear that you got it working. I don't use phpmyadmin myself either. I can recommend DBeaver for database work. That's my favorite :)

**Comment by otalado** [comment]
I used DBeaver myself too.I just feel it is a bit of overkill for my usage.
As the matter of fact I still keep it in my Linux PC, but I don't use it
very often as I don't have any serious project on it lately...

V sob., 27. jul. 2024 16:55 je oseba Andreas Kristiansen <
***@***.***> napisala:
